pysat/_instrument.py

```diff
@@ -1888,6 +1974,212 @@ def concat_data(self, new_data, prepend=False, **kwargs):
         self.data = concat_func(new_data, **kwargs)
         return
 
+    def custom_attach(self, function, kind='modify', at_pos='end', args=[],
+                      kwargs={}):
+        """Attach a function to custom processing queue.
+
+        Custom functions are applied automatically whenever `.load()`
+        command called.
+
+        Parameters
+        ----------
+        function : string or function object
+            name of function or function object to be added to queue
+        kind : {'add', 'modify', 'pass'}
+            - add
+                Adds data returned from function to instrument object.
+                A copy of pysat instrument object supplied to routine.
+            - modify
+                pysat instrument object supplied to routine. Any and all
+                changes to object are retained.
+            - pass
+                A copy of pysat object is passed to function. No
+                data is accepted from return.
+            (default='modify')
+
+        at_pos : string or int
+            Accepts string 'end' or a number that will be used to determine
+            the insertion order if multiple custom functions are attached
+            to an Instrument object. (default='end').
+        args : list or tuple
+            Ordered arguments following the instrument object input that are
+            required by the custom function (default=[])
+        kwargs : dict
+            Dictionary of keyword arguments required by the custom function
+            (default={})
+
+        Note
+        ----
+        Allowed `attach` function returns:
+
+        - {'data' : pandas Series/DataFrame/array_like,
+          'units' : string/array_like of strings,
+          'long_name' : string/array_like of strings,
+          'name' : string/array_like of strings (iff data array_like)}
+
+        - pandas DataFrame, names of columns are used
+
+        - pandas Series, .name required
```

Comment:
What about xarray Datasets?

Reply:
I added support for this type in the latest commits.